packages/netlify-cms-core/src/reducers/collections.ts

```diff
@@ -30,10 +30,9 @@ function collections(state = defaultState, action: ConfigAction) {
   switch (action.type) {
     case CONFIG_SUCCESS: {
       const collections = action.payload.collections;
-      const newState: { [name: string]: CmsCollection } = {};
-      for (const collection of collections) {
-        newState[collection.name] = collection;
-      }
+      const newState = Object.fromEntries(
```

Comment:
Can't agree that using map + fromEntries is more readable then simple loop :) functional style doesn't really shine here and looks way more confusing then the loop.

Reply:
Actually both methods have a bug of not maintaining the order of the collections as appeared in the configuration.
See https://github.com/netlify/netlify-cms/pull/4960/commits/c0e43bc9ff84a7b0cf260d4f72c38d2f29ffd391

Reply:
Yeah, in future we can use js Map to maintain the order

Reply:
But I'm wondering, why is it happening in the first place. I thought modern js engines maintain property order in js objects. It wasn't the case in the past, but I was sure that it had been fixed.

Reply:
Yeah, I'm pretty sure this is the problem with Immutable. This works correctly. The order is being maintained.

```js
let obj = {}; 
let arr = ['collection_1', 'collection_3', 'collection_2']
for (const item of arr) {
  obj[item] = item;
}
```

Though the order is not being maintained if property name is a number.

Reply:
Yes, JS keeps insertions order. It's the `fromJS` that doesn't know how to handle it.